packages/block-library/src/block/edit.js

```diff
@@ -1,180 +1,161 @@
-			}
-		} );
+	componentDidUpdate( prevProps ) {
+		if ( this.props.reusableBlock !== prevProps.reusableBlock ) {
+			this.setState( { reusableBlockInstanceId: this.state.reusableBlockInstanceId + 1 } );
+		}
 	}
 
-	setTitle( title ) {
-		this.setState( { title } );
+	toggleIsEditing( isEditing ) {
+		this.setState( { isEditing } );
 	}
 
-	save() {
-		const { reusableBlock, onUpdateTitle, updateAttributes, block, onSave } = this.props;
-		const { title, changedAttributes } = this.state;
+	saveEdit() {
+		this.toggleIsEditing( false );
 
-		if ( title !== reusableBlock.title ) {
-			onUpdateTitle( title );
-		}
+		const {
+			getEditedPostAttribute,
+			getEditedPostContent,
+			getBlocks,
+		} = this.registry.select( 'core/editor' );
+
+		const reusableBlock = {
+			...this.props.reusableBlock,
+			title: getEditedPostAttribute( 'title' ),
+			content: {
+				raw: getEditedPostContent(),
+			},
+		};
 
-		updateAttributes( block.clientId, changedAttributes );
-		onSave();
+		const [ parsedBlock ] = getBlocks();
 
-		this.stopEditing();
+		this.props.onSave( reusableBlock, parsedBlock );
 	}
 
 	render() {
-		const { isSelected, reusableBlock, block, isFetching, isSaving } = this.props;
-		const { isEditing, title, changedAttributes } = this.state;
+		const { setIsSelected, reusableBlock, isSelected, isSaving } = this.props;
+		const { settingsWithLock, isEditing, reusableBlockInstanceId } = this.state;
 
-		if ( ! reusableBlock && isFetching ) {
+		if ( ! reusableBlock ) {
 			return <Placeholder><Spinner /></Placeholder>;
 		}
 
-		if ( ! reusableBlock || ! block ) {
-			return <Placeholder>{ __( 'Block has been deleted or is unavailable.' ) }</Placeholder>;
-		}
-
-		let element = (
-			<BlockEdit
-				{ ...this.props }
-				isSelected={ isEditing && isSelected }
-				clientId={ block.clientId }
-				name={ block.name }
-				attributes={ { ...block.attributes, ...changedAttributes } }
-				setAttributes={ isEditing ? this.setAttributes : noop }
-			/>
-		);
-
+		let list = <BlockList />;
 		if ( ! isEditing ) {
-			element = <Disabled>{ element }</Disabled>;
+			list = <Disabled>{ list }</Disabled>;
 		}
 
 		return (
-			<Fragment>
-				{ element }
-				{ ( isSelected || isEditing ) && (
-					<ReusableBlockEditPanel
-						isEditing={ isEditing }
-						title={ title !== null ? title : reusableBlock.title }
-						isSaving={ isSaving && ! reusableBlock.isTemporary }
-						onEdit={ this.startEditing }
-						onChangeTitle={ this.setTitle }
-						onSave={ this.save }
-						onCancel={ this.stopEditing }
-					/>
-				) }
-				{ ! isSelected && ! isEditing && <ReusableBlockIndicator title={ reusableBlock.title } /> }
-			</Fragment>
+			<RegistryProvider value={ this.registry }>
+				<EditorProvider
+					// Force a rerender when reusableBlock is updated, so that changes made
+					// externally appear in the nested editor.
+					key={ reusableBlockInstanceId }
```

Comment:
Not very proud of this. Ideally, `EditorProvider` would update when its `post` prop changes but the `SETUP_EDITOR` effect makes this difficult...